### ISSUE TITLE
Search whitelist when typing

### ISSUE THREAD

**Issue by rugk** [Cookie-AutoDelete/Cookie-AutoDelete#120]
If one types something into the input box in the whitelist settings, the add-on should search for that string in the whitelist and filter the results, so that one can see if there is already a entry for a particular website.

Also this makes the input box there a "dual-purpose" thing as one can then use it to search *and* to add entries.

**Comment by mrdokenny** [comment]
I was thinking it should filter your cookie domains instead. Say you type in 'g', it would filter the cookie domains with 'g' like google.com

**Comment by rugk** [comment]
What are "cookie domains"?

All domains, which once set a cookie? If so, that makes few sense, it would basically just be your browser history, i.e. you can also use your address bar for that.

**Comment by mrdokenny** [comment]
Let me rephrase it. It would function more like auto-complete this way. Type in 'g', it would give a list of 'g' related cookies that you have right now. So it would be a quicker way of adding domains based on the cookies you have.

**Comment by rugk** [comment]
Hmm, you could do, but then a search field for searching the whitelist would still be good.

What about doing both things. Showing a autocomplete-drop-down directly below the input (similar to how search engines do it) and still filtering the whitelist (which is even below that), so one can click anywhere to hide the auto-complete field and then have the filtered list?

**Comment by kennethtran93** [comment]
Initial implementation will only search expressions/hostnames.  Think I'll leave searching for cookie-specific data to other extensions.

Closing as resolved from #941
